zubhub_frontend/zubhub/src/views/signup/signupScripts.js

```diff
@@ -214,32 +214,67 @@ export const setLabelWidthOfStaticFields = (refs, document, props) => {
  * @todo - describe object's function
  */
 export const validationSchema = Yup.object().shape({
+  role: Yup.string().required('Please choose a role'),
   username: Yup.string().required('required'),
-  email: Yup.string()
-    .email('invalid')
-    .test('email_is_empty', 'phoneOrEmail', function (value) {
-      return vars.email_field_touched && !value && !this.parent.phone
-        ? false
-        : true;
-    }),
-  phone: Yup.string()
-    .test('phone_is_invalid', 'invalid', function () {
-      return vars.iti.isValidNumber() || !vars.iti.getNumber() ? true : false;
-    })
-    .test('phone_is_empty', 'phoneOrEmail', function () {
-      return vars.phone_field_touched &&
-        !vars.iti.getNumber() &&
-        !this.parent.email
-        ? false
-        : true;
-    }),
-  dateOfBirth: Yup.date().max(new Date(new Date().getFullYear() - 12, 1, 1), 'oldEnough').required('required'),
-  user_location: Yup.string().min(1, 'min').required('required'),
-  password1: Yup.string().min(8, 'min').required('required'),
-  password2: Yup.string()
-    .oneOf([Yup.ref('password1'), null], 'noMatch')
+  email: Yup.string().email('invalid').required('Add your email'),
+  // .test('email_is_empty', 'phoneOrEmail', function (value) {
+  //   return vars.email_field_touched && !value && !this.parent.phone ? false : true;
+  // }),
+  phone: Yup.string(),
+  // .test('phone_is_invalid', 'invalid', function () {
+  //   return vars.iti.isValidNumber() || !vars.iti.getNumber() ? true : false;
+  // })
+  // .test('phone_is_empty', 'phoneOrEmail', function () {
+  //   return vars.phone_field_touched && !vars.iti.getNumber() && !this.parent.email ? false : true;
+  // }),
+  dateOfBirth: Yup.date().required('required'),
+  location: Yup.string().min(1, 'min').required('required'),
+  password: Yup.string().min(8, 'min').required('required'),
+  confirmPassword: Yup.string()
+    .oneOf([Yup.ref('confirmPassword'), null], 'noMatch')
     .required('required'),
   bio: Yup.string().max(255, 'tooLong'),
+  parentEmail: Yup.string().email(),
+  parentPhone: Yup.string(),
+  gender: Yup.string().required('Choose your gender'),
+  sendTips: Yup.boolean(),
 });
 
 // /^[+][0-9]{9,15}$/g.test(value)
+
+export const formikSchema = {
+  initialValues: {
+    dateOfBirth: '',
+    role: '',
+    username: '',
+    email: '',
+    phone: '',
+    location: '',
+    password: '',
+    confirmPassword: '',
+    bio: '',
+    parentEmail: '',
+    parentPhone: '',
+    gender: null,
+    sendTips: false,
+  },
+  validationSchema,
+};
+
+export const customValidation = (values, props) => {
```

Comment:
this should not be here, this should be part of the validation above. We should not have a different custom validation